### DISCUSSION TITLE
write response to file

### DISCUSSION THREAD

**Discussion by reisman234** [thunderclient/thunder-client-support#890] [Ideas]
I make request against an API which send larger files back, to prevent that everything goes into the response output window. It would be nice to have an Option to save the response body from the request directly to a file.

**Comment by rangav** [comment]
There is vscode setting, which control response size, above that limit, it will hide the response and show save to file
<img width="400" alt="Screenshot 2022-11-22 at 09 01 31" src="https://user-images.githubusercontent.com/8637550/203271402-635d0406-bc7d-41bf-8bf5-e47df4ea2867.png">

You can also save any response to file

<img width="455" alt="Screenshot 2022-11-22 at 09 01 45" src="https://user-images.githubusercontent.com/8637550/203271573-9fa8c15d-f064-4136-b70f-a9153ff71cd1.png">

**Comment by rangav** [comment]
I added more size options in `v2.0.1`, and change the default size to `2MB`

<img width="399" alt="Screenshot 2022-11-24 at 16 29 47" src="https://user-images.githubusercontent.com/8637550/203831352-46aadd99-44c2-4251-9049-7feac349700c.png">

**Comment by reisman234** [comment]
I tested now the first download with a smaller file, which comes not near that that Response Limit.

At first, it seems the data is correct, but to verify that, I want to compare it with the origin file. I tried to use `Save to File` Button but nothing happens, is this a bug?  

![image](https://user-images.githubusercontent.com/44891585/203852612-0cfc95d1-5bf3-4de4-a6c5-467a5f37f180.png)

**Reply by rangav** [reply]
It looks like png image, can you share response headers screenshot?

**Reply by reisman234** [reply]
At the moment, I took not much effort what the headers are, maybe that could be the missing point. 

![image](https://user-images.githubusercontent.com/44891585/203856794-0ecec66e-e47c-4ece-9d7f-493e33e82994.png)

**Reply by rangav** [reply]
There is no content-type header in response, that’s why it’s not displaying.

**Reply by rangav** [reply]
Do you have test url so I can test and see if there is any fix for this?

**Reply by reisman234** [reply]
I think the content type is really missing, when I test it with an public available image it makes a preview and the `save to file` button works.

https://static.wikia.nocookie.net/desimpsons/images/1/14/Ralph_Wiggum.png/revision/latest?cb=20151015180621

![image](https://user-images.githubusercontent.com/44891585/203860111-c8bb73ee-f1f9-4164-a749-c96b049e5cbf.png)

**Comment by reisman234** [comment]
I tested now the download of a larger binary file (~1Gi), but now the request stuck in a processing state. On my Resource monitor I can see the download activity, but after that ends nothing happen (sill processing, also after 30 min).

The header which I send via uvicorn is: 

{'Accept-Ranges': 'bytes', 'Content-Length': '1241985027', 'Content-Security-Policy': 'block-all-mixed-content', 'Content-Type': 'application/octet-stream', 'ETag': '"3c22ffe6ee8c8d5c364eca47238804f5-75"', 'Last-Modified': 'Wed, 23 Nov 2022 20:28:02 GMT', 'No-Gzip-Compression': 'true', 'Strict-Transport-Security': 'max-age=31536000; includeSubDomains', 'Vary': 'Origin, Accept-Encoding', 'X-Amz-Request-Id': '172AC89C2E590B1E', 'X-Content-Type-Options': 'nosniff', 'X-Xss-Protection': '1; mode=block', 'Date': 'Fri, 25 Nov 2022 09:06:25 GMT'}

The download with curl and the "--output" parameter works without problems.

The green field is the network activity
![image](https://user-images.githubusercontent.com/44891585/203948145-f01e7fc0-5290-4d00-9212-43049520f7fb.png)

**Reply by rangav** [reply]
Are you downloading 1GB file? Not sure this API Client works with very large files. 

Is it part of API testing? or just the using it for all use-cases.

**Reply by reisman234** [reply]
For me, it is just a test for my API, which returns larger binary files.

**Reply by rangav** [reply]
Ok got it, Please create a new feature request, will add to roadmap

**Reply by reisman234** [reply]
ok, I will do that!
